system/Autoloader/Autoloader.php

```diff
@@ -373,7 +373,16 @@ private function loadComposerNamespaces(ClassLoader $composer, array $composerPa
             unset($namespacePaths['CodeIgniter\\']);
         }
 
-        $packageList = InstalledVersions::getAllRawData()[0]['versions'];
+        if (method_exists(InstalledVersions::class, 'getAllRawData')) {
+            // This method requires Composer 2.0.14 or later.
+            $packageList = InstalledVersions::getAllRawData()[0]['versions'];
+        } else {
+            throw new RuntimeException(
+                'Your Composer version is too old.'
+                . ' Please update Composer (run `composer self-update`) and remove your vendor/ directory,'
```

Comment:
Give more information in Exception

```suggestion
                . ' Please update Composer (run `composer self-update`) to v2.0.14 or later and remove your vendor/ directory,'
```

Reply:
Done.